### ISSUE TITLE
ERROR:Dummy prediction failed with run state StatusType.CRASHED

### ISSUE THREAD

**Issue by gugugu-469** [automl/Auto-PyTorch#391]
* **I'm submitting a ...**
  - [x] error report
  - [ ] bug report
  - [ ] feature request
  - [ ] support request => Please do not submit support request here, see note at the top of this template.

# Issue Description
I use TabularClassificationTask() on a large dataset which has 54162 nums of train data , each data has 20000 features.The warning message "Time limit for a single run is higher than total time limit." seems to appear all the time even I set the total_walltime_limit to 1e20. And the code can't work with the ERROR below.

## Your Code

```python

import h5py
from autoPyTorch.api.tabular_classification import TabularClassificationTask

# data and metric imports
import sklearn.model_selection
import sklearn.datasets
import sklearn.metrics

import numpy as np

with h5py.File('./events20000/events20000/event_prediction_dataset20000.h5') as dfile:
    X = dfile['X'][:]
    Y = dfile['Y'][:]
    Y=np.argmax(Y,axis=-1)

X_train, X_test, y_train, y_test = \
        sklearn.model_selection.train_test_split(X, Y, random_state=1)

cls = TabularClassificationTask()


# Search for an ensemble of machine learning algorithms
cls.search(
    X_train=X_train,
    y_train=y_train,
    X_test=X_test,
    y_test=y_test,
    optimize_metric='accuracy',
    memory_limit=None,
    total_walltime_limit=1e50
)

# Calculate test accuracy
y_pred = cls.predict(X_test)
score = cls.score(y_pred, y_test)
print("Accuracy score", score)
```

## Error message

```
[ERROR] [2022-02-27 20:02:00,532:Client-pynisher] Cannot recover the output from /tmp/tmpyqasocq9/std.out due to [Errno 2] No such file or directory: '/tmp/tmpyqasocq9/std.out'
[ERROR] [2022-02-27 20:02:00,532:Client-pynisher] Cannot recover the output from /tmp/tmpyqasocq9/std.err due to [Errno 2] No such file or directory: '/tmp/tmpyqasocq9/std.err'
[DEBUG] [2022-02-27 20:02:00,535:Client-pynisher] Redirecting output of the function to files. Access them via the stdout and stderr attributes of the wrapped function.
[DEBUG] [2022-02-27 20:02:01,695:Client-TAE] Finish function evaluation 1.
Status: StatusType.CRASHED, Cost: 1.0, Runtime: 0.10478663444519043,
Additional information:
error: Result queue is empty
exit_status: 5
subprocess_stdout: None
subprocess_stderr: None
exitcode: 1
configuration_origin: DUMMY
[ERROR] [2022-02-27 20:02:01,696:Client-AutoPyTorch:fd7952a8-97c3-11ec-bd55-63965984cecd:1] Dummy prediction failed with run state StatusType.CRASHED and additional info:
error: Result queue is empty
exit_status: 5
subprocess_stdout: None
subprocess_stderr: None
exitcode: 1
configuration_origin: DUMMY.
```

**Comment by gugugu-469** [comment]
When I decrease the total_walltime_limit to 1000 , the code can run with another ERROR appears ```[ERROR] [2022-02-28 10:14:05,597:Client-AutoPyTorch:5b56f63e-9839-11ec-ae95-63965984cecd:1] Traditional prediction for lgb failed with run state StatusType.TIMEOUT.```. But it seems the algorithm doesn't work , the accuracy score is only 0.005, and I found the GPU memory usage is very low ,it's only 255 MiB. I wonder if I have some settings that are not set properly.

**Comment by nabenabe0928** [comment]
Hi, I am sorry to hear that, but could you describe your system information?
I am now suspecting that your system was down due to the too large dataset.

If I understand your post correctly, your data has 54K data points x 20K feature columns ≒ 1G elements.
If all the columns are 64 bit float, the dataset itself consumes 8GB, so depending on your system environment, it just crashes and thus we get CRASH.
Please correct me if I am wrong!

**Comment by gugugu-469** [comment]
Thanks @nabenabe0928 ! I think you are right , and the code works when I test the code again on another dataset which has 54K x 400 feature columns. But it seems unable to find a model , the result of the show_models function is None . I set the parameters as below ```memory_limit=None,total_walltime_limit=500,min_budget=30,max_budget=50```.  If I should increase the value of total_walltime_limit or any other parameters to get the correct result?
I also find the usage of the GPU memory is only 256MiB every thread , I could only increase the number of n_jobs to increase the utilization rate of GPU. I wonder if it is normal?

**Comment by ravinkohli** [comment]
Hi @gugugu-469, you can find out how the search went by using the [`sprint_statistics`](https://github.com/automl/Auto-PyTorch/blob/06e67de5017b4cccad9398e24a3d9f0bd8176da3/autoPyTorch/api/base_task.py#L1455) function. It's possible that none of the configurations was successfully trained.

Also, are you storing the logs? If not, I would recommend you to make a run by passing the tmp and output directories while instantiating the `TabularClassificationTask` you can find more information [here](https://github.com/automl/Auto-PyTorch/blob/06e67de5017b4cccad9398e24a3d9f0bd8176da3/examples/20_basics/example_tabular_classification.py#L43-L46). 

If you are storing the logs, you can find more information on what is causing this issue. It would be useful if you could provide us with those details.

**Comment by gugugu-469** [comment]
Thanks @ravinkohli ! I get a model when I reset the ```total_walltime_limit``` parameter to 2000. But I am also trouble with the usage of GPU. I wonder if it is normal that the usage of the GPU memory is only 256MiB every thread.

**Comment by ravinkohli** [comment]
Actually, we have not tested our code to be run with GPUs yet but we could try to debug if you could provide more details of the script you are running for example the environment variables that you are setting, the number of threads you are passing to autopytorch, etc. You shouldn't face this problem if you run on a CPU.

**Comment by gugugu-469** [comment]
I just set like this
 ```os.environ["CUDA_VISIBLE_DEVICES"]="0"
os.environ['JOBLIB_TEMP_FOLDER'] = tmp.gettempdir()
os.environ['OMP_NUM_THREADS'] = '10'
os.environ['OPENBLAS_NUM_THREADS'] = '10'
os.environ['MKL_NUM_THREADS'] = '10'
cls = TabularClassificationTask(n_jobs=10)
```

I intend to shorten the time of searching by using GPU ( I don't know whether it is possible) . And I find the usage of the GPU memory is only 256MiB every thread.
